services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/GoalStateManagerTest.java

```diff
@@ -0,0 +1,4101 @@
+            + "               \"priority\":0,\n"
+            + "               \"associatedType\":\"MAIN\",\n"
+            + "               \"associatedTableId\":\"ece52017-ac10-442f-b50a-7f058ca16164\",\n"
+            + "               \"projectId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "               \"id\":\"360a7a8e-31da-4cab-95e5-7bfff271f604\",\n"
+            + "               \"name\":\"default_route_rule\",\n"
+            + "               \"description\":\"\"\n"
+            + "            }\n"
+            + "         ],\n"
+            + "         \"gatewayMacAddress\":\"AA-BB-CC-E0-30-98\",\n"
+            + "         \"ipVersion\":4,\n"
+            + "         \"ipV4RangeId\":\"e2845526-62bd-4c1f-a16d-e20ec665b536\",\n"
+            + "         \"revisionNumber\":1,\n"
+            + "         \"tenantId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "         \"dnsPublishFixedIp\":false,\n"
+            + "         \"useDefaultSubnetpool\":false,\n"
+            + "         \"created_at\":\"2020-06-24 21:58:05\",\n"
+            + "         \"updated_at\":\"2020-06-24 21:58:05\",\n"
+            + "         \"projectId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "         \"id\":\"9192a4d4-ffff-4ece-b3f0-8d36e3d88000\"\n"
+            + "      }\n"
+            + "   ],\n"
+            + "   \"securityGroups\":[\n"
+            + "      {\n"
+            + "         \"tenantId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "         \"securityGroupRules\":[\n"
+            + "            {\n"
+            + "               \"tenantId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "               \"securityGroupId\":\"3dda2801-d675-4688-a63f-dcda8d111111\",\n"
+            + "               \"direction\":\"egress\",\n"
+            + "               \"etherType\":\"IPv4\",\n"
+            + "               \"id\":\"8111156c-2336-4f41-aae9-338c32b76a09\"\n"
+            + "            },\n"
+            + "            {\n"
+            + "               \"tenantId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "               \"securityGroupId\":\"3dda2801-d675-4688-a63f-dcda8d111111\",\n"
+            + "               \"direction\":\"egress\",\n"
+            + "               \"etherType\":\"IPv6\",\n"
+            + "               \"id\":\"99414929-e045-44e3-951f-e5ac0d14006f\"\n"
+            + "            }\n"
+            + "         ],\n"
+            + "         \"createAt\":\"2020-06-24 15:02:39\",\n"
+            + "         \"updateAt\":\"2020-06-24 15:02:39\",\n"
+            + "         \"projectId\":\"3dda2801-d675-4688-a63f-dcda8d327f50\",\n"
+            + "         \"id\":\"3dda2801-d675-4688-a63f-dcda8d111111\",\n"
+            + "         \"name\":\"sg1\",\n"
+            + "         \"description\":\"string\"\n"
+            + "      }\n"
+            + "   ]\n"
+            + "}";
+    String output =
```

Comment:
We might want to do the same change for Output as for input. Not high priority though. This is for our own readability and better maintenance.

Reply:
fixed in 55a056a